federation/pkg/kubefed/init/init.go

```diff
@@ -292,7 +304,7 @@ func (i *initFederation) Run(cmdOut io.Writer, config util.AdminConfig) error {
 
 	fmt.Fprint(cmdOut, "Creating federation control plane service...")
 	glog.V(4).Info("Creating federation control plane service")
-	svc, ips, hostnames, err := createService(cmdOut, hostClientset, i.commonOptions.FederationSystemNamespace, serverName, i.commonOptions.Name, i.options.apiServerAdvertiseAddress, i.options.apiServerServiceType, i.options.dryRun)
+	svc, ips, hostnames, err := createService(cmdOut, hostClientset, i.commonOptions.FederationSystemNamespace, serverName, i.commonOptions.Name, i.options.apiServerAdvertiseAddress, i.options.apiServerNodePortsPort, i.options.apiServerServiceType, i.options.dryRun)
```

Comment:
I am not sure what you have in mind here. The "port != 0" check is used to add a nodePort to the port on the svc request https://github.com/kubernetes/kubernetes/pull/46283/files#diff-e7243d86cc50e209d78ce51d57dad48dR464 The second place where we check the provided port is there where we validate the input parameters https://github.com/kubernetes/kubernetes/pull/46283/files#diff-e7243d86cc50e209d78ce51d57dad48dR235 

I am a bit confused as to what this pointer would point to especially given the check conditions are not exactly the same.

Can you please elaborate a bit on what needs to be done and I will do it :)

Thank you for the review